### DISCUSSION TITLE
Style individual tooltips

### DISCUSSION THREAD

**Discussion by BionicBlade0** [Alexays/Waybar#3734] [Q&A]
Dear Altruists,
Is it possible to style the tooltip for an individual module separately from the other tooltips? If yes, how?
Thanks in advance.

**Comment by xynapz** [comment]
@BionicBlade0 did you do it? style individual module's tooltip? if so could you share it.

**Reply by BionicBlade0** [reply]
I don't think there is any way to do it in waybar. i ended up coding my own bar using quickshell and qml.

**Reply by xynapz** [reply]
i couldn't find it anywhere as well.

**Comment by jsabutis** [comment]
Not with CSS, and the reason is structural rather than a missing feature. GTK3 does not draw the tooltip inside the module. It hands the text to one shared tooltip window owned by the display, which is a separate toplevel, so the module's node is never an ancestor of it. That is why `#clock tooltip {}` or `#pulseaudio tooltip label {}` silently match nothing, while a bare `tooltip {}` and `tooltip label {}` work: those are the only nodes in that tree.

You can still differentiate per module, just through the text instead of the stylesheet. Waybar sets tooltip content with `set_markup`, so Pango markup in `tooltip-format` is live:

```jsonc
"clock": {
  "tooltip-format": "<span size='large' foreground='#89b4fa'><b>{:%A %d %B}</b></span>\n<tt>{calendar}</tt>"
},
"pulseaudio": {
  "tooltip-format": "<span font_family='JetBrainsMono Nerd Font' size='small'>{desc}</span>"
}
```

`size`, `foreground`, `background`, `weight`, `font_family`, `style` and `strikethrough` all work inside those spans, so per module type, colour and emphasis are reachable.

What stays global is everything at box level: the tooltip's own background, padding, border and radius come from that single `tooltip {}` rule and there is no per module hook for them. If one module genuinely needs a different tooltip shape, the honest answer is that it has to stop being a GTK tooltip and become your own popup.
